include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -3590,6 +3590,14 @@ struct OrtApi {
   */
   void(ORT_API_CALL* MemoryInfoGetDeviceType)(_In_ const OrtMemoryInfo* ptr, _Out_ OrtMemoryInfoDeviceType* out);
 
+  /* \brief Update the OrtEnv instance with custom log severity level
+   *
+   * \param[in] log_severity_level The log severity level.
+   * \param[in] ort_env The OrtEnv instance being used
+   *
+   * \since Version 1.13.
+   */
+  ORT_API2_STATUS(UpdateEnvWithCustomLogLevel, OrtLoggingLevel log_severity_level, _In_ const OrtEnv* ort_env);
```

Comment:
You can't really have OrtEnv const when you've an API to update it :-).

Reply:
Thanks for pointing out! That has been fixed

Reply:
Please, add C++ API so users do not have to go to C back and forth.

Reply:
C++ API has been added in [latest commit ](https://github.com/microsoft/onnxruntime/pull/13402/commits/9715a16b2b77b5a6ff59326c981c433e66b9c0d5)as well. Thanks!

Reply:
One important thing to mention is thread-safety. Typically, we do not update the environment after the creation and any thread or even multiple sessions can use it.

Reply:
> One important thing to mention is thread-safety. Typically, we do not update the environment after the creation and any thread or even multiple sessions can use it.

Really appreciate the context you shared. The reason adding this env_update feature is this feature exists in Pybind but missing in c# binding, and we would like to unify the user experience and provide the ability to adjust default logger

This feature reuses the existing design in [Pybind:set_default_logger_severity](https://github.com/microsoft/onnxruntime/blame/952c99304a764328070ac6c07f510deec06dedf4/onnxruntime/python/onnxruntime_pybind_state.cc#L923-L924). Do you have any suggestions in order to improve this design?